src/main/java/org/sonar/plugins/stash/StashIssueReportingPostJob.java

```diff
@@ -21,96 +24,118 @@
-          String repository = stashRequestFacade.getStashRepository();
-          String stashPullRequestId = stashRequestFacade.getStashPullRequestId();
-
-          int stashTimeout = config.getStashTimeout();
-
-          StashCredentials stashCredentials = stashRequestFacade.getCredentials();
-
-          try (StashClient stashClient = new StashClient(stashURL, stashCredentials, stashTimeout, config.getSonarQubeVersion())) {
-
-              StashUser stashUser = stashRequestFacade.getSonarQubeReviewer(stashCredentials.getLogin(), stashClient);
 
-              if (stashUser == null) {
-                  throw new StashMissingElementException("no SonarQube reviewer identified"
-                          + " to publish to Stash the SQ analysis");
-              }
-
-              // Get all changes exposed from Stash differential view of the pull-request
-              StashDiffReport diffReport = stashRequestFacade.getPullRequestDiffReport(stashProject, repository,
-                      stashPullRequestId, stashClient);
-              if (diffReport == null) {
-                  throw new StashMissingElementException("No Stash differential report"
-                          + " available to process the SQ analysis");
-              }
-
-              // if requested, reset all comments linked to the pull-request
-              if (config.resetComments()) {
-                  stashRequestFacade.resetComments(stashProject, repository, stashPullRequestId,
-                          diffReport, stashUser, stashClient);
-              }
-
-              boolean canApprovePullrequest = config.canApprovePullRequest();
-              if (canApprovePullrequest) {
-                  stashRequestFacade.addPullRequestReviewer(stashProject, repository, stashPullRequestId,
-                          stashCredentials.getLogin(), stashClient);
-              }
-
-              // if threshold exceeded, do not push issue list to Stash
-              if (issueReport.countIssues() >= issueThreshold) {
-                  LOGGER.warn("Too many issues detected ({}/{}): Issues cannot be displayed in Diff view",
-                          issueReport.countIssues(), issueThreshold);
-              } else {
-                  stashRequestFacade.postCommentPerIssue(stashProject, repository, stashPullRequestId, sonarQubeURL,
-                          issueReport, diffReport, stashClient);
-              }
-
-              stashRequestFacade.postAnalysisOverview(stashProject, repository, stashPullRequestId, sonarQubeURL,
-                      issueThreshold, issueReport, stashClient);
-
-              // if no new issues, plugin approves the pull-request
-              if (canApprovePullrequest && issueReport.countIssues() == 0) {
-                  stashRequestFacade.approvePullRequest(stashProject, repository, stashPullRequestId,
-                          stashCredentials.getLogin(), stashClient);
+    StashClient stashClient = null;
+    try {

```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Change this "try" to a try-with-resources. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2093)